### ISSUE TITLE
Annotations of pre-annotated data don't get saved after update

### ISSUE THREAD

**Issue by mariambasents** [heartexlabs/label-studio#1790]
**Describe the bug**
I uploaded pre-annotated text data into Label Studio. The annotators are either adding new annotation by selecting text region, then updating, or editing and changing the pre-annotations and updating. Also there is a Binary Choice ("yes", "no") which has default value (the default is "no"). 
In any case the updates often don't get saved for this project. Other projects work fine. The labeling interface code doesn't have any errors, it's shown in the "To Reproduce"  section. We tried clearing browser cookies and cache, working in incognito mode but it didn't help. Changing annotator's account helped temporarily but the problem returned. 
I saw in the issues that almost similar problem happened before (https://github.com/heartexlabs/label-studio/issues/1101) but it didn't get solved since the author stopped doing the labeling. The author stated that they "decided it was mostly due to poor internet connections" but it doesn't hold in our case.

**To Reproduce**
The behavior happens not constantly and it's not clear what causes it to happen. I guess you need to do a lot of labeling and then the problem will arise.
The labeling interface code
```
<View>
  <Labels name="label_expansion" toName="google_expansion" showInline="true" choice="multiple">
    <Label value="Filler" background="#ADA49E" granularity="symbol" hint="for google expansion text highlighting"/>
  </Labels>
  <Labels name="label" toName="original" showInline="true" choice="multiple">
    <Label value="EXPN" background="#FFA39E" granularity="symbol" hint="abbreviations"/>
    <Label value="CARDINAL" background="#FFC069" granularity="symbol"/>
    <Label value="DECIMAL" background="#D3F261" granularity="symbol"/>
    <Label value="FRACTION" background="#F759AB" granularity="symbol"/>
    <Label value="ORDINAL" background="#5CDBD3" granularity="symbol"/>
    <Label value="DIGIT" background="#9254DE" granularity="symbol"/>
    <Label value="DATE" background="#ADC6FF" granularity="symbol"/>
    <Label value="TIME" background="#096DD9" granularity="symbol"/>
    <Label value="MONEY" background="#FFC069" granularity="symbol"/>
    <Label value="MEASURE" background="#FFA39E" granularity="symbol"/>
    <Label value="PERCENT" background="#AD8B00" granularity="symbol"/>
    <Label value="RANGE" background="#389E0D" granularity="symbol"/>
    <Label value="LETTERS" background="#D4380D" granularity="symbol"/>
    <Label value="TELEPHONE" background="#FFA39E" granularity="symbol"/>
    <Label value="ADDRESS" background="#9254DE" granularity="symbol"/>
    <Label value="URL" background="#389E0D" granularity="symbol"/>
    <Label value="EMAIL" background="#389E0D" granularity="symbol"/>
    <Label value="HASHTAG" background="#5CDBD3" granularity="symbol"/>
    <Label value="VERBATIM" background="#5CDBD3" granularity="symbol"/>
    <Label value="WDLK" background="#FFC069" granularity="symbol" hint="misspelling"/>
  </Labels>
  <View style="margin-top: 20px; padding: 20px; box-shadow: 2px 2px 8px #AAA;">
    <Header value="Original text"/>
    <Text name="original" value="$original"/>
  </View>
  <View visibleWhen="region-selected">
    <Header value="Write the expansion"/>
    <TextArea name="expansion" toName="original" editable="true" perRegion="true" required="false"/>
  </View>
  <View style="margin-top: 20px; padding: 20px; box-shadow: 2px 2px 8px #AAA;">
    <Header value="Google dataset expansion"/>
    <Text name="google_expansion" value="$google_expansion"/>
  </View>
  <View style="margin-top: 20px; padding: 20px; box-shadow: 2px 2px 8px #AAA;">
    <Header value="Normalise expansion"/>
    <Text name="norm_expansion" value="$norm_expansion"/>
  </View>
  <View style="margin-top: 10px; margin-bottom: 10px; padding: 20px; box-shadow: 2px 2px 8px #AAA;">
    <Choices name="type" toName="original" choice="single">
      <Header value="Is the original text problematic?"/>
      <Choice value="Yes"/>
      <Choice value="No"/>
    </Choices>
  </View>
</View>
```

1. Upload pre-annotated text data. There is an example.
{'id': '11',
 'data': {'original': 'On October 25, 2014 , they held their 500th sit in protest .',
  'google_expansion': 'On october twenty fifth twenty fourteen , they held their five hundredth sit in protest .',
  'norm_expansion': 'On october twenty fifth twenty fourteen, they held their five hundredth sit in protest.'},
 'annotations': [{'id': '397425',
   'completed_by': {'id': 1,
    'email': 'some_email',
    'first_name': 'name',
    'last_name': 'name'},
   'was_cancelled': False,
   'ground_truth': False,
   'created_at': '2021-52-23T16:52:12',
   'updated_at': '2021-52-23T16:52:12',
   'lead_time': 0,
   'result_count': 0,
   'prediction': {},
   'result': [{'id': '397425_5201099',
     'from_name': 'label',
     'to_name': 'original',
     'type': 'labels',
     'value': {'start': 3,
      'end': 19,
      'text': 'October 25, 2014',
      'labels': ['DATE']}},
    {'id': '397425_5201099',
     'from_name': 'expansion',
     'to_name': 'original',
     'type': 'textarea',
     'value': {'start': 3,
      'end': 19,
      'text': ['october twenty fifth twenty fourteen']}},
    {'id': '397425_52010990',
     'from_name': 'label_expansion',
     'to_name': 'google_expansion',
     'type': 'labels',
     'value': {'start': 3,
      'end': 39,
      'text': 'october twenty fifth twenty fourteen',
      'labels': ['DATE']}},
    {'id': '397425_5201104',
     'from_name': 'label',
     'to_name': 'original',
     'type': 'labels',
     'value': {'start': 38,
      'end': 43,
      'text': '500th',
      'labels': ['ORDINAL']}},
    {'id': '397425_5201104',
     'from_name': 'expansion',
     'to_name': 'original',
     'type': 'textarea',
     'value': {'start': 38, 'end': 43, 'text': ['five hundredth']}},
    {'id': '397425_52011040',
     'from_name': 'label_expansion',
     'to_name': 'google_expansion',
     'type': 'labels',
     'value': {'start': 58,
      'end': 72,
      'text': 'five hundredth',
      'labels': ['ORDINAL']}},
    {'id': 'choices_692603',
     'from_name': 'type',
     'to_name': 'original',
     'type': 'choices',
     'value': {'choices': ['No']}}]}]}
2. Select a label and region or click on the pre-annotated region.
3. Write an expansion for that region or edit the expansion of the pre-annotated region.
4. Click on Update.
5. Repeat 2-4 steps multiple times (maybe several days).
6. The updates will start to not get saved.

**Expected behavior**
Task annotations are getting saved after updates.

**Screenshots**
The logs when the saving problem happened. 
<img width="1203" alt="image" src="https://user-images.githubusercontent.com/86047912/143826267-9b341939-4977-4676-ba8c-880d94db0a43.png">

**Environment (please complete the following information):**
 - OS: Ubuntu (deployment), macOS, Windows 10 (users)
 - Label Studio Version: 1.3.post0

**Comment by makseq** [comment]
@mariambasents 
1 Could you reproduce it on your side constantly
2 Could you check your browser console too? 
3 Are you sure POST/PATCH api/annotations/ is called when you lose you data?

**Comment by mariambasents** [comment]
@makseq 
1. It's not happening constantly. 
2. Browser console logs after the problem occurred
[LS_browser_logs.log](https://github.com/heartexlabs/label-studio/files/7641835/LS_browser_logs.log)
3. I am not sure but I will try to check.

**Comment by makseq** [comment]
@mariambasents Another idea - could you pay attention to drafts? maybe a previously made draft overwrites the correct annotation?

**Comment by mariambasents** [comment]
@makseq
Thanks, we will pay more attention to that. 
Also the following happened - the annotator changed the pre-annotation of one of the tasks but on the page nothing got changed. At that time we checked the same task with another account and turned out that the change was there. In this case the saving worked but it was not shown on annotator's page.

**Comment by noircir** [comment]
Similar problem here. When a pre-annotated file is loaded, the changes are intermittently not saved after pressing Update. They are not saved **visually** for the annotator. It is hard to know if they were saved in the background, or not. When the editing happens on a server, the other users can see the change. The editor cannot see the change even after re-login. If the editor creates a new account, then it is possible to see the change from the new account (it is not clear though if having one new account is enough, or if the editor should keep creating new accounts just to see the changes).  Editing pre-annotated file locally (where no other users could check), sometimes having a private browser window helps to see the changes, but not always. 

See attached window how the changes are not saved visually. 

https://user-images.githubusercontent.com/22533804/160938418-d938d15d-f0c0-4a36-99be-0288c081a1c2.mp4

**Comment by makseq** [comment]
@noircir Thank you for your video. This is related to drafts, try to click on it after update.

**Comment by noircir** [comment]
Agree. I would rather see my edited work after pressing Update. And not the other way around.

**Comment by makseq** [comment]
@noircir https://github.com/heartexlabs/label-studio/pull/2185 --- could you please check this PR? This issue should be fixed there.

**Comment by noircir** [comment]
@makseq Sorry for a stupid question, but how do I check it? Should I download a new version? And where to get it?

**Comment by makseq** [comment]
```
# Clone repo
git clone https://github.com/heartexlabs/label-studio.git

git fetch origin DEV-1556
git checkout DEV-1556

# Install dependencies
cd label-studio
pip install -e .

# Collect static files
python label_studio/manage.py collectstatic

# Launch
python label_studio/manage.py runserver
```

**Comment by noircir** [comment]
<img width="812" alt="Screen Shot 2022-04-02 at 8 32 08 PM" src="https://user-images.githubusercontent.com/22533804/161406126-d586a5db-ab85-4dd4-ad1a-713547d3242e.png">

**Comment by makseq** [comment]
```
# Activate virtual environment
python -m venv env
source env/bin/activate

# Clone repo
git clone https://github.com/heartexlabs/label-studio.git

cd label-studio

git fetch origin DEV-1556
git checkout DEV-1556

# Install dependencies
cd label-studio
pip install -e .

# Collect static files
python label_studio/manage.py collectstatic

# Launch
python label_studio/manage.py runserver

```

**Comment by noircir** [comment]
( in the above code, the second `cd label-studio` is not needed )

<img width="1748" alt="Screen Shot 2022-04-05 at 8 50 56 AM" src="https://user-images.githubusercontent.com/22533804/161757525-e4b9588c-4367-45fd-96fd-67d2f6bae4b7.png">

**Comment by noircir** [comment]
Traceback (most recent call last):
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/backends/sqlite3/base.py", line 413, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.OperationalError: no such column: task.updated_by_id

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/elena/working_dir/label-studio/label_studio/data_import/api.py", line 355, in post
    return super(ReImportAPI, self).post(*args, **kwargs)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/utils/decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "/Users/elena/working_dir/label-studio/label_studio/data_import/api.py", line 176, in post
    return super(ImportAPI, self).post(*args, **kwargs)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/rest_framework/generics.py", line 190, in post
    return self.create(request, *args, **kwargs)
  File "/Users/elena/working_dir/label-studio/label_studio/core/utils/common.py", line 334, in f_retry
    return f(*args, **kwargs)
  File "/Users/elena/working_dir/label-studio/label_studio/data_import/api.py", line 319, in create
    project.remove_tasks_by_file_uploads(file_upload_ids)
  File "/Users/elena/working_dir/label-studio/label_studio/projects/models.py", line 420, in remove_tasks_by_file_uploads
    self.tasks.filter(file_upload_id__in=file_upload_ids).delete()
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/models/query.py", line 746, in delete
    collector.collect(del_query)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/models/deletion.py", line 243, in collect
    new_objs = self.add(objs, source, nullable,
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/models/deletion.py", line 107, in add
    if not objs:
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/models/query.py", line 291, in __bool__
    self._fetch_all()
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/models/query.py", line 1308, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/models/query.py", line 53, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1156, in execute_sql
    cursor.execute(sql, params)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/sentry_sdk/integrations/django/__init__.py", line 545, in execute
    return real_execute(self, sql, params)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/elena/working_dir/env/lib/python3.9/site-packages/django/db/backends/sqlite3/base.py", line 413, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: no such column: task.updated_by_id

**Comment by makseq** [comment]
Run this before `runserver`:
```
python label_studio/manage.py migrate
```

**Comment by noircir** [comment]
Yes it works. The functionality is now more user-friendly: change an annotation -> Update -> the changed annotation is showing. I am not sure if the original author here is satisfied, but I am.

**Comment by hlomzik** [comment]
fixed in #2185
